src/replit_river/error_schema.py

```diff
@@ -17,9 +17,16 @@
 # ERROR_CODE_CANCEL is the code used when either server or client cancels the stream.
 ERROR_CODE_CANCEL = "CANCEL"
 
+# ERROR_CODE_SESSION_CLOSED is the code used when either server or client closes
+# the session.
+ERROR_CODE_SESSION_CLOSED = "CLOSED"
```

Comment:
this is not in the protocol, UNEXPECTED_DISCONNECT is probably the closest semantically?

Reply:
We're not sending this over the wire, we are forced to give a reason to the python code calling this method. `UNEXPECTED_DISCONNECT` doesn't make sense in the case where we intentionally closed the session but a consumer still tried to call a method post-close

Reply:
im fine leaving it as is, maybe just make the doc comment more clear this is a synthetic message that isnt from the server

Reply:
Yeah also caught my eye. Also fine either way, we haven't had a situation where we needed to differentiate in the TS side because things are pretty automatic as far as re-sending goes. We had to differentiate in crosis because we didn't have the right API to make retries automatic-as-needed.

Reply:
by automatic i mean it's transaction-like, because the consumer is typically react, things rerun in a transaction-like fashion.

Reply:
I think the primary footgun is that there's no distinct error channel for these synthetic errors, so I overcommunicated here. If there end up being more places where we need more synthetic errors we can create a distinct channel for them, distinct from the `RiverServiceException` hierarchy.